### ISSUE TITLE
 Cannot print after serializing Python module

### ISSUE THREAD

**Issue by LiuChao888** [uqfoundation/dill#444]
**I have a py file named test1.py  with only one simple method like this:**

```
def a1():
    print("OK")
```

**I have another py file named test2.py, import test1 in test2.py, the code is as follows:**

```
import sys
import dill
from test1 import a1


if __name__ == "__main__":
    sys.modules.pop("test1")
    a2 = dill.loads(dill.dumps(a1))
    a2()
```

**When I serialize, print cannot be executed, and the error message is as follows:**

```
Traceback (most recent call last):
  File "test2.py", line 9, in <module>
    a2()
  File "C:\Users\codes\test\test1.py", line 3, in a1
    print("OK")
NameError: name 'print' is not defined
```

I checked the ID of the module object in sys.modules before and after serialization, the ID has changed, but I don't know if it has something to do with this

**Comment by mmckerns** [comment]
I can't reproduce what you are seeing.  What version of python and dill are you using? I've tried python 2.7, and 3.6 - 3.11 and `dill` `0.3.5.dev0`

**Comment by LiuChao888** [comment]
> I can't reproduce what you are seeing. What version of python and dill are you using? I've tried python 2.7, and 3.6 - 3.11 and `dill` `0.3.5.dev0`

I was using Python 3.6 and dill 0.3.3
But now I'm using dill 0.3.4, this problem doesn't appear anymore, but I don't know why?

**Comment by mmckerns** [comment]
I'm assuming it's due to a bug that was fixed in `0.3.4`... so, I'm closing this.  @LiuChao888: please feel free to reopen it if needed.
